### ISSUE TITLE
Install locations and features missed on Debian

### ISSUE THREAD

**Issue by jaraco** [pypa/distutils#2]
In pypa/setuptools#2232, we learned that Ubuntu has a customized version of distutils, so when Setuptools attempts to adopt distutils from the standard library, those customizations are lost and Ubuntu users are unhappy.

I propose to fold ~~[those patches (3.8)](https://sources.debian.org/patches/python3.8/3.8.4%7Erc1-1/distutils-install-layout.diff/)~~ [those patches](https://sources.debian.org/patches/python3.9/3.9.2-1/distutils-install-layout.diff/) into this project, enabling Setuptools to incorporate that functionality. This approach will need to be done in a way that only takes effect selectively on those systems.

**Comment by pradyunsg** [comment]
> I propose to fold [those patches](https://sources.debian.org/patches/python3.8/3.8.4%7Erc1-1/distutils-install-layout.diff/) into this project, enabling Setuptools to incorporate that functionality. This approach will need to be done in a way that only takes effect selectively on those systems.

As discussed in the linked issue, we're taking a different approach now with pypa/setuptools#2232 -- re-enabling the functionality with the known degradation for Debian systems.

This prevents locking-out Debian users from the general/unpatched behavior of distutils, while users who are using the distro-provided-and-patched python can/should use older versions of setuptools (or the version provided by their distro) to avoid breakage.

**Comment by jaraco** [comment]
As Debian-specific support should come from Debian and site-customizations is happening in #12, this issue is essentially superseded by #12.

**Comment by jaraco** [comment]
I've decided to revisit this issue. I desperately want to re-enable `SETUPTOOLS_USE_DISTUTILS=local` as the default in Setuptools and provide a means for all users to migrate off of distutils in the stdlib. I can't do that without breaking Debian and as of yet, I've seen no mechanism that will enable Debian to transition from their current aggressive patches to something else.

So what I'm planning to do is, in addition to allowing for the customized schemes, also provide hooks that will enable the current set of customizations. In the [feature/debian-compat branch](/pypa/distutils/tree/feature/debian-compat), I've managed to refactor the code to enable this change.

My plan now is to provide a hook that happens as part of `import distutils` that would allow for applying the customizations.
